src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBaseKernels.hpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file FlowSolverBaseKernels.hpp
+ */
+
+#ifndef GEOSX_PHYSICSSOLVERS_FLUIDFLOW_FLOWSOLVERBASEKERNELS_HPP
+#define GEOSX_PHYSICSSOLVERS_FLUIDFLOW_FLOWSOLVERBASEKERNELS_HPP
+
+#include "common/DataTypes.hpp"
+#include "finiteVolume/FluxApproximationBase.hpp"
+#include "rajaInterface/GEOS_RAJA_Interface.hpp"
+#include "linearAlgebra/interfaces/InterfaceTypes.hpp"
+
+namespace geosx
+{
+
+namespace FlowSolverBaseKernels
+{
+
+/******************************** PorosityKernel ********************************/
+struct PorosityKernel
+{
+  template< typename POLICY, typename POROSITY_WRAPPER >
+  static void
+  launch( localIndex const size,
+          POROSITY_WRAPPER const & porWrapper,
+          arrayView1d< real64 const > const & pres,
+          arrayView1d< real64 const > const & dPres )
+  {
+    forAll< POLICY >( size, [=] GEOSX_HOST_DEVICE ( localIndex const k )
+    {
+      for( localIndex q = 0; q < porWrapper.numGauss(); ++q )
+      {
+        porWrapper.update( k, q, pres[k] + dPres[k] );
+      }
+    } );
+  }
+
+};
+
+/******************************** PermeabilityKernel ********************************/
+
+template< typename REGIONTYPE >
+struct PermeabilityKernel
+{};
+
+template<>
+struct PermeabilityKernel< CellElementSubRegion >
+{
+  template< typename POLICY, typename PERM_WRAPPER >
+  static void
+  launch( localIndex const size,
+          PERM_WRAPPER const & permWrapper,
+          arrayView2d< real64 const > const & porosity )
+  {
+    forAll< POLICY >( size, [=] GEOSX_HOST_DEVICE ( localIndex const k )
+    {
+      for( localIndex q = 0; q < permWrapper.numGauss(); ++q )
+      {
+        permWrapper.update( k, q, porosity[k][q] );
+      }
+    } );
+  }
```

Comment:
Do you anticipate future permeability models that will not be written as a function of porosity only, and do you think it will be easy to modify this if we encounter a model that does not fit? Recently, we were came across two use cases for which the highlighted lines above may need some changes:
- Permeability (and porosity) tabulated directly as a function of pressure. For permeability, this is probably not something that we want inside GEOSX (?)
- Permeability tabulated as a function of strain

Reply:
So, my plan was to add at least an interface for the poroelastic case in which the permeability is not a function of the porosity but of the strain. I would have a separate interface for this case though. I think that it makes sense to assume that for flow-only problems the strain is not registered and does not exist. Basically I was planning on having some `PoroelasticFlowKernels` to assemble fluxes and update `permeability` and `porosity` for the `poroelastic` case. I know that it means having some more code rewriting but I think that it keeps our assembly kernels much cleaner and allows us to avoid polluting simple tpfa flow kernels with things that are only relevant for the coupled problems.